src/pages/patientView/sampleManager.tsx

```diff
@@ -18,23 +17,71 @@ export function placeArrow(tooltipEl: any) {
 }
 
 
-// sort samples based event, clinical data and id
+// sort samples based on event, clinical data and id
+// 1. based on sample collection data (timeline event)
+// 2. if all cases have derived normalized case types, put primary first
+// 3. natural sort of sample ids
 export function sortSamples(samples: Array<ClinicalDataBySampleId>,
-                            clinicalDataLegacyCleanAndDerived: { [s:string]:any }) {
-    // sort by
-    // 1. based on sample collection data (timeline event) (unimplemented)
-    // 2. if all cases have derived normalized case types, put primary first (unimplemented)
-    // 3. natural sort of sample ids
-    let sortedSampleIDs: string[] = [];
-    sortedSampleIDs = sortedSampleIDs.concat(samples.map((sample) => sample.id)).sort(naturalSort).reverse();
-
-    return _.sortBy(samples, function(sample) {
-        return sortedSampleIDs.indexOf(sample.id);
+                            clinicalDataLegacyCleanAndDerived: { [s:string]:any },
+                            events?: any) {
+    // natural sort
+    let naturalSortedSampleIDs: string[] = [];
+    naturalSortedSampleIDs = naturalSortedSampleIDs.concat(samples.map((sample) => sample.id)).sort(naturalSort);
+
+    // based on sample collection data (timeline event)
+    let collectionDayMap: {[s:string]:number} = {};
+    if (events) {
+        let specimenEvents = events.filter((e: ClinicalEvent) => (e.eventType === 'SPECIMEN'));
+
+        for (let i: number = 0; i < specimenEvents.length; i++) {
+            let sampleAttr = _.find(specimenEvents[i].attributes, (attr: ClinicalEventData) => {
+                // TODO: This is legacy support for old timeline data that does not use SAMPLE_ID, but one of the specrefnum
+                return (attr.key === "SAMPLE_ID" || attr.key === "SpecimenReferenceNumber" || attr.key === "SPECIMEN_REFERENCE_NUMBER") &&
+                    (naturalSortedSampleIDs.indexOf(attr.value) !== -1);
+            });
+            if (sampleAttr && sampleAttr.value !== null && sampleAttr.value !== undefined) {
```

Comment:
i think the first condition actually covers both the subsequent null and undefined checks.